### PR TITLE
breaking changes incoming

### PR DESCRIPTION
**Pull request by arseniybelkov**
# Breaking Changes
- `MetricLogger` is `MetricMonitor` from now on.
# Fixes
- `TimeProfiler` reloads state
- `MetricMonitor` doesn't throw error with `Incosistent losses` after experiment is rerun.
- Docs minor improvements.
#  New Features
- `WandbLogger` has flag `allow_rerun`. It allows to log metrics to the same experiment after experiment restart.
- `MetricMonitor` now has `preprocessing` for `single metrics`

### CODE REVIEW

File: thunder/policy.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def get_lr(self) -> List[float]:
         return juxt(self.current_mapping)(self.last_epoch)
 
     def state_dict(self) -> Dict[str, Any]:
-        return super().state_dict("mapping")
+        return super().state_dict("mapping", "current_mapping")
```
Comment (maxme1) [inline thunder/policy.py:166]: doesn't this mean that the signature should be `state_dict(self, *args)`?

Comment (maxme1) [reply thunder/policy.py:166]: then why you're calling super().state_dict with some args?

File: thunder/policy.py [hunk 2]
```diff
@@ -163,7 +160,7 @@ def get_lr(self) -> List[float]:
         return juxt(self.current_mapping)(self.last_epoch)
 
     def state_dict(self) -> Dict[str, Any]:
-        return super().state_dict("mapping")
+        return super().prepare_state_dict("mapping", "current_mapping")
```
Comment (maxme1) [inline thunder/policy.py:163]: you're not overloading prepare_state_dict, so no need for `super`

File: thunder/policy.py [hunk 3]
```diff
@@ -193,8 +190,5 @@ def get_lr(self) -> List[float]:
             for param_group, mapping in zip_equal(self.optimizer.param_groups, self.current_mapping)
         ]
 
-    def state_dict(self) -> Dict[str, Any]:
-        return super().state_dict()
-
     def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         super().load_state_dict(state_dict)
```
Comment (maxme1) [inline thunder/policy.py:194]: what's this for?

Comment (arseniybelkov) [reply thunder/policy.py:194]: load_state_dict is abstract, needs to be overriden

Comment (maxme1) [reply thunder/policy.py:194]: then why do you call super here? does it have some default logic?

Comment (arseniybelkov) [reply thunder/policy.py:194]: It does

**Review by maxme1 — APPROVED** [review]

**Pull request merged**